.github/workflows/automationTemplate.yaml

```diff
@@ -29,6 +29,10 @@ env:
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions: 
+      actions: write
+      contents: write
+      pull-requests: write
```

Comment:
nit: just curious as to the impact defining this `permissions` property has on the workflow

Reply:
By explicitly calling the permissions, it'll override the organizational setting on the github action permissions just for this specific workflow. By default, microsoft prevents the usage of actions outside of the org and prevents actions open PR. So just to be cautious, i put the permissions needed for this workflow.

actions permission allows this workflow to cancel or trigger another workflow
contents allow this workflow to commit things
pull-requests allow it to create/modify PRs